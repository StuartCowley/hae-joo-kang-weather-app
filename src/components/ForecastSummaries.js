import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

const ForecastSummaries = ({ forecasts, onForecastSelect }) => {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon.toString()}
          temperature={forecast.temperature}
          onSelect={onForecastSelect}
        />
      ))}
    </div>
  );
};

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.number,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }),
      onSelect: PropTypes.func,
    })
  ).isRequired,
  onForecastSelect: PropTypes.func.isRequired,
};

export default ForecastSummaries;
