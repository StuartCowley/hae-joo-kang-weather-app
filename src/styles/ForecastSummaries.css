.forecast-summaries {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-around;
}

@media (min-width: 768px) {
  .forecast-summaries {
  flex-direction: row;
  }
}