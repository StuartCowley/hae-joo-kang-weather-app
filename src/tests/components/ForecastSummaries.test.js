import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1525046400000,
      description: "Stub description",
      icon: "800",
      temperature: {
        min: 4,
        max: 11,
      },
      onSelect: () => {},
    },
    {
      date: 1525132800000,
      description: "Stub description2",
      icon: "211",
      temperature: {
        min: 8,
        max: 13,
      },
      onSelect: () => {},
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(<ForecastSummaries forecasts={validProps} />);
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
