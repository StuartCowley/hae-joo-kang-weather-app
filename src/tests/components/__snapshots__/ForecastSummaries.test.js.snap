// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Mon Apr 30 2018
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        11°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Tue May 01 2018
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        13°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description2
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
